src/libs/actions/IOU.ts

```diff
@@ -1684,14 +1686,14 @@ function getDeleteTrackExpenseInformation(
         });
     }
 
-    if (actionableWhisperReportActionID) {
+    if (actionableWhisperReportActionID && actionableWhisperReportAction?.actionName === CONST.REPORT.ACTIONS.TYPE.ACTIONABLE_TRACK_EXPENSE_WHISPER) {
```

Comment:
Why do we need the additional check? I think this conditional check is not required if we fetch the whisper action inside the `if` block here.